src/module.c

```diff
@@ -3677,7 +3679,12 @@ void moduleTypeNameByID(char *name, uint64_t moduleid) {
  *   'when' argument is either REDISMODULE_AUX_BEFORE_RDB or REDISMODULE_AUX_AFTER_RDB.
  * * **aux_load**: A callback function pointer that loads out of keyspace data from RDB files.
  *   Similar to aux_save, returns REDISMODULE_OK on success, and ERR otherwise.
- *
+ * * **free_effort**: A callback function pointer that used to determine whether the module's
```

Comment:
I see this is a direct translation from OS, but I think the notion of free_effort is unnecessary and confusing from a module context. I feel like a better callback would be, can_free_async(), which can make it's own decision about whether or not this is worth freeing async.

Reply:
@madolson what do you mean by "OS"?
we're trying to match the effort to:
```
#define LAZYFREE_THRESHOLD 64
```
i think a name like `can_free_async` is misleading (it's possible that it "can" but better not do that). if we're seeking a boolean return, maybe a better name would be `should_free_async`.
but then if we some day let users control this constant via command or config, the modules will need to be able to query it.
i think returning the complexity (how many pointers you have to free, or how many loops you'll need to run) makes sense as a return value.
i suppose we better document that 0 indicate not to be released by a thread.

Reply:
Open source? I didn't know what to call none module redis code, redis core maybe?

I still think should free async makes a more sense. The lazy free estimate is an approximation for "should we free this async". A module doesn't need to make the same decision we do.

I also think we should never expose the lazy free effort as a parameter, without some type of conversion in between. You could be more explicit and pass the effort to the module callback function and it can choose if it wants to ignore it. I just think we should have more thoughtful apis. From an outside user it's very hard to understand.